### ISSUE TITLE
Allow `cur_svy()` to work inside `mutate()`?

### ISSUE THREAD

**Issue by bschneidr** [gergness/srvyr#138]
I'm working on a pull request to try and implement the `cur_svy_wts()` function suggested in #136, and as a result I noticed that `cur_svy()` works inside `summarize.tbl_svy()` but not inside `mutate.tbl_svy()`.

``` r
library(srvyr)

library(survey)
data(api)

# Create a survey design object
dstrata <- apistrat %>%
  as_survey_design(strata = stype, weights = pw)

# Able to access weights in summarize
dstrata %>%
  summarize(sum_of_wts = sum(1/cur_svy()$prob))
#> # A tibble: 1 x 1
#>   sum_of_wts
#>        <dbl>
#> 1      6194.

# But not in mutate
dstrata %>%
  mutate(wts = 1/cur_svy()$prob)
#> Error: Problem with `mutate()` column `wts`.
#> i `wts = 1/cur_svy()$prob`.
#> x Survey context not set
```

<sup>Created on 2021-12-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Is there a design reason why `cur_svy()` isn't supported inside `mutate()`? If not, I'm happy to see if we can get it to work.

**Comment by gergness** [comment]
Yeah that makes sense. I just hadn’t thought of a situation where you’d want the survey context set in a mutate, maybe needs it for all dplyr verbs (filtering by weights too?)

**Comment by gergness** [comment]
Filtering by comparing to the mean in a single step is probably a more likely situation

**Comment by bschneidr** [comment]
Thanks, Greg. That's a good common example. Another more specialized reason for filtering on the weights is doing diagnostics of nonresponse adjustments. For example, you might want to look at cases where the adjusted weights are much different than the original weights. Or you might want to look at cases with especially large or small weights so that you can trim them to try to reduce variances.

Do you think it would make sense to add context for `mutate()` and `filter()`? I don't really see a downside other than hassle of implementation, but maybe I'm missing something. If you think it makes sense, I'm happy to put together a pull request for it.

**Comment by gergness** [comment]
Yep, makes sense! Thanks! I think it’s as simple as adding two lines like this, but I may be forgetting something 
https://github.com/gergness/srvyr/blob/main/R/summarise.r#L6

**Comment by ray-p144** [comment]
I just ran into this today. I was trying to figure out how many missing values there were for the group with non-zero weights so I tried to use `cur_svy()` in `filter()`. It didn't work, so I tried using it in `mutate()` and that didn't work either.

I logged on to GitHub to see if there were any open issues on it just to find that you guys are already on top of it!

**Comment by gergness** [comment]
@bschneidr - did you get a chance to work on this? I'm hoping for some time to look at srvyr in the next week or two, would be nice to get this in the release. Happy to take a look myself if you haven't.

**Comment by bschneidr** [comment]
Yep, sorry for the delayed reply @gergness. I had to take some time off of open-source work the last few weeks but finally have time now to work on this.

**Comment by gergness** [comment]
closed via #139
